python/kserve/kserve/protocol/dataplane.py

```diff
@@ -116,6 +123,25 @@ async def live() -> Dict[str, str]:
         Returns:
             Dict: {"status": "alive"}
         """
+        # If predictor host is present, then it means this is a transformer,
+        # We should also need to check the predictor server's health if predictor health check is enabled.
+        if DataPlane.predictor_health_check and DataPlane.predictor_host:
+            if DataPlane.predictor_protocol == PredictorProtocol.GRPC_V2.value:
+                grpc_client = get_grpc_client(DataPlane.predictor_host, DataPlane.predictor_use_ssl)
+                res = await grpc_client.ServerLive(pb.ServerLiveRequest())
+                if not res.live:
+                    return {"status": "error"}
+            elif (DataPlane.predictor_protocol == PredictorProtocol.REST_V1.value or
+                  DataPlane.predictor_protocol == PredictorProtocol.REST_V2.value):
+                res = await get_http_client().get(get_liveness_endpoint(DataPlane.predictor_host,
```

Comment:
We can enhance the `inference_client.py` to support the liveness and readiness client API